tests/test_datasets/test_dataset.py

```diff
@@ -190,6 +191,20 @@ def test_get_data_with_nonexisting_class(self):
         # Check that no label is mapped to 3, since it is reserved for label '4'.
         self.assertEqual(np.sum(y == 3), 0)
 
+    def test_get_data_corrupt_pickle(self):
+        # lazy loaded dataset, populate cache
+        self.iris.get_data()
+        # corrupt pickle file, overwrite as empty.
+        with open(self.iris.data_pickle_file, "w") as fh:
+            fh.write("")
+
+        with catch_warnings():
```

Comment:
Could you please also check for the error message? I find it way easier to to parse the code then when I know what error message is tested for.

Reply:
I removed this check since we now use the logger instead of raising an actual warning.